debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/KafkaConnectRecordParser.java

```diff
@@ -189,54 +189,72 @@ protected void parseKeyFields(Struct key, Record r) {
-                    // in the default configuration of the stream, for an update, the fields in the after struct
-                    // are only the delta fields, therefore, it is possible for a field to not be there.
-                    if (valueAndSet == null){
-                        continue;
-                    }
-                } else if (r.op.equals("d") && valueAndSet == null){
-                    // in case of deletes we are using before struct which contains only delta for yb
+            // TODO: write a proper transformer for this logic
+            // values in the debezium connector are as follows:
+            // "val1" : {
+            //  "value" : "value for val1 column",
+            //  "set" : true
+            //}
+            Struct valueAndSet = after.getStruct(f.name());
+            if (r.op.equals("u")) {
+                // in the default configuration of the stream, for an update, the fields in the after struct
+                // are only the delta fields, therefore, it is possible for a field to not be there.
+                if (valueAndSet == null){
                     continue;
                 }
-                
-                if (!valueAndSet.getBoolean("set")){
+            } else if (r.op.equals("d") && valueAndSet == null){
+                // in case of deletes we are using before struct which contains only delta for yb
+                continue;
+            }
+            
+            if (!valueAndSet.getBoolean("set")){
+                continue;
+            }
+            Object afterFieldValue = valueAndSet.getWithoutDefault("value");
+            r.addAfterValueField(f.name(), afterFieldValue);
+        }
+    }
+
+    private void parseValueFieldsForOthers(Struct after, Struct before, Record r) {
+        if (r.op.equals("d")){ // after is null for delete events
+            for (Field f : before.schema().fields()) {
+                Object beforeFieldValue = before.getWithoutDefault(f.name());
+                r.addBeforeValueField(f.name(), beforeFieldValue);
+            }
+            return;
+        }
+
+        for (Field f : after.schema().fields()) {
+            if (r.op.equals("u")) {
+                if (Objects.equals(after.get(f), before.get(f))) {
+                    // no need to record this as field is unchanged
                     continue;
                 }
-                fieldValue = valueAndSet.getWithoutDefault("value");
             }
-            else{
-                if (r.op.equals("u")) {
-                    if (Objects.equals(after.get(f), before.get(f))) {
-                        // no need to record this as field is unchanged
-                        continue;
-                    }
-                }
-                fieldValue = after.getWithoutDefault(f.name());
+            Object afterFieldValue = after.getWithoutDefault(f.name());
+            Object beforeFieldValue = null;
+            if (!r.op.equals("c") && before != null) { 
```

Comment:
How about just check `if before != null` here, and then validate above before line 242 that for update events, before struct should be non-null?

Reply:
I changed that to:
```
if (!(r.op.equals("c") || r.op.equals("r"))){ 
                // before is null for create events
                // before can also be null if REPLICA IDENTITY is not set as 'FULL' (maybe in case of snapshot-only mode)
                beforeFieldValue = before.getWithoutDefault(f.name());
}
```

We can have a simple check of `before != null` but i thought this will be more meaningful and easier to understand in code.